client/rpc.go

```diff
@@ -301,7 +300,7 @@ func (c *Client) rpcConnListener() {
 
 // listenConn is used to listen for connections being made from the server on
 // pre-existing connection. This should be called in a goroutine.
-func (c *Client) listenConn(s *yamux.Session) {
+func (c *Client) listenConn(s *pool.Conn) {
```

Comment:
Tiniest of nitpicks, but with the type being changed to `*pool.Conn`, naming this argument `s` doesn't make sense anymore. Maybe rename the arg to `p`?

(`pool.Conn` is kind of a weirdly named type in general because it's more like "connection factory" or "connect proxy" given that `Accept()` returns `net.Conn`, but it isn't a "connection pool" either as it contains the connection pool. But probably best not to rework the whole thing. 😀 )

Reply:
I'll change it to `c`.  FWIW, `pool.Conn` is the underlying physical connection - `pool.Accept()` returns a wrapped `yamux.Session` which also implements `net.Conn` (and clients expect)..